lib/generators/module/module_generator.rb

```diff
@@ -5,49 +5,66 @@
 class ModuleGenerator < Rails::Generators::NamedBase
   source_root File.expand_path('templates', __dir__)
 
-  def create_app
-    path = "modules/#{file_name}/app"
-    template 'app/controllers/controller.rb.erb',
-             File.join(path, 'controllers', file_name, 'v0', "#{file_name}_controller.rb")
-    template 'app/controllers/application_controller.rb.erb',
-             File.join(path, 'controllers', file_name, 'application_controller.rb')
-    template 'app/models/resource.rb.erb', File.join(path, 'models', file_name, 'resource.rb')
-    template 'app/serializers/serializer.rb.erb',
-             File.join(path, 'serializers', file_name, "#{file_name}_serializer.rb")
-    template 'app/services/configuration.rb.erb', File.join(path, 'services', file_name, 'configuration.rb')
-    template 'app/services/service.rb.erb', File.join(path, 'services', file_name, 'service.rb')
+  def create_directory_structure
+    # create the dir structure here
+    %w[controllers models serializers service].each do |dir|
+      FileUtils.mkdir_p "modules/#{file_name}/app/#{dir}" unless Dir.exist?("modules/#{file_name}/app/#{dir}")
+    end
   end
 
-  def create_lib
+  def create_engine
+    # create engine file
     path = "modules/#{file_name}/lib"
     template 'lib/namespace/engine.rb.erb', File.join(path, file_name, 'engine.rb')
```

Comment:
Does this support someone typing `rails generate module potato` and then the engine class properly name it `Potato`?  The template doesn't seem to change the casing.  I think we should consider lazy users.  Now that I'm typing, that should be a new ticket for iterating on this.

Reply:
Correct, the engine template will use the class_name and name it Potato https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/generators/module/templates/lib/namespace/engine.rb.erb#L3

The above file was unchanged, so that's why it's not in the changed files